File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int n = matrix.size();
        int m = matrix[0].size();
        unordered_set<int> row;
        unordered_set<int> col;
        
        for(int i = 0; i < n; i++){
            for(int j = 0 ; j< m ; j++){
                if(matrix[i][j]==0){
                    row.insert(i);
                    col.insert(j);
                }
            }
        }
        for(int i = 0 ;i< n;i++){
            if(row.count(i)){
                for(int j = 0; j < m; j++){
                    matrix[i][j] = 0;
                }
            }
        }
        
        for(int j=0;j<m;j++){
            if(col.count(j)){
                for(int i=0;i<n;i++){
                    matrix[i][j]=0;
                }
            }
        }

    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;

        boolean[] row = new boolean[m];
        boolean[] col = new boolean[n];

        for (int r = 0; r < m; r++) {
            for (int c = 0; c < n; c++) {
                if (matrix[r][c] == 0) {
                    row[r] = true;
                    col[c] = true;
                }
            }
        }

        for (int r = 0; r < m; r++) {
            for (int c = 0; c < n; c++) {
                if (row[r] || col[c]) {
                    matrix[r][c] = 0;
                }
            }
        }
    }
}